cypress/integration/plugins/security-dashboard-plugin/aggregation_view.js

```diff
@@ -0,0 +1,352 @@
+  
+      cy.url().should((url) => {
+        expect(url).to.contain('/users/create');
+      });
+  
+      cy.contains('span', 'Create');
+  
+      cy.get('input[data-test-subj="name-text"]').type(userName2, {
+        force: true,
+      });
+      cy.get('input[data-test-subj="name-text"]').should(
+        'have.value',
+        userName2
+      );
+  
+      cy.get('input[data-test-subj="password"]').type(password, {
+        force: true,
+      });
+      cy.get('input[data-test-subj="re-enter-password"]').type(password, {
+        force: true,
+      });
+  
+      cy.contains('button', 'Create').first().click({ force: true });
+  
+      cy.url().should((url) => {
+        expect(url).to.contain('/users');
+      });
+  
+      cy.contains('h3', 'Internal users');
+      // should contain the new user that was just created
+      cy.contains('a', userName2);
+    });
+
+    it('should create new role with test tenant permission successfully', () => {
+      cy.visit(SEC_UI_ROLES_CREATE_PATH);
+  
+      cy.contains('span', 'Create');
+  
+      cy.get('input[data-test-subj="name-text"]').type(roleName1, {
+        force: true,
+      });
+  
+      cy.get('input[data-test-subj="name-text"]').should(
+        'have.value',
+        roleName1
+      );  
+      cy.get('input[id="cluster-permission-box"]').type(clusterPermission);
+      cy.get('button[title="cluster_all"]').click();  
+      cy.get('input[id="index-input-box"]').type(indexName);  
+      cy.get('input[id="index-permission-box"]').type(indexPermission);
+      cy.get('button[title="indices_all"]').click();
+      
+      cy.get('input[id="tenant-permission-box"]').type(tenantName);
+      cy.get('button[title="test"]').click();
+  
+      cy.contains('button', 'Create').first().click({ force: true });
+  
+      cy.url().should((url) => {
+        expect(url).to.contain('/roles/view/');
+      });
+  
+      cy.contains('h1', roleName1);
+    });
+
+    it('should create new role without test tenant permission successfully', () => {
+      cy.visit(SEC_UI_ROLES_CREATE_PATH);
+  
+      cy.contains('span', 'Create');
+  
+      cy.get('input[data-test-subj="name-text"]').type(roleName2, {
+        force: true,
+      });
+  
+      cy.get('input[data-test-subj="name-text"]').should(
+        'have.value',
+        roleName2
+      );  
+      cy.get('input[id="cluster-permission-box"]').type(clusterPermission);
+      cy.get('button[title="cluster_all"]').click();  
+      cy.get('input[id="index-input-box"]').type(indexName);  
+      cy.get('input[id="index-permission-box"]').type(indexPermission);
+      cy.get('button[title="indices_all"]').click();
+  
+      cy.contains('button', 'Create').first().click({ force: true });
+  
+      cy.url().should((url) => {
+        expect(url).to.contain('/roles/view/');
+      });
+  
+      cy.contains('h1', roleName2);
+    });
+
+    it('should map the user "test1" to role "roleWithTest"', () => {
```

Comment:
looks like the following 4 tests are same except for using different path and username, could you please make a list of data, and run them one by one in a for loop?
you can find multiple examples in [Dynamic tests from Cypress](https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/fundamentals__dynamic-tests), for example, [this one](https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/fundamentals__dynamic-tests/cypress/e2e/subdomains-spec.cy.js) is pretty neat